File: Python For Everyone, Horstmann/Chapter3-Decisions/P3.13.py
# Write a program that translates a number between 0 and 4 into the closest letter
# grade. For example, the number 2.8 (which might have been the average of several
# grades) would be converted to B–. Break ties in favor of the better grade; for exam ple
# 2.85 should be a B.

gradeInput = float(input("Enter your grade: "))

gradeOutput = ""

if gradeInput > 4 or gradeInput < 0:
    exit("Invalid grade input. Grades are between 4.0 and 0.0")

if gradeInput == 4 or gradeInput >= 3.85:
    gradeOutput = "A"

elif gradeInput < 3.85 and gradeInput >= 3.7:
    gradeOutput = "A-"

elif gradeInput == 3.3 or gradeInput > 3:
    gradeOutput = "B+"

elif gradeInput == 3 or gradeInput >= 2.85:
    gradeOutput = "B"

elif gradeInput < 2.85 and gradeInput >= 2.7:
    gradeOutput = "B-"

elif gradeInput == 2.3 and gradeInput > 2:
    gradeOutput = "C+"

elif gradeInput == 2 or gradeInput >= 1.85:
    gradeOutput = "C"

elif gradeInput < 1.85 and gradeInput >= 1.7:
    gradeOutput = "C-"

elif gradeInput == 1.3 and gradeInput > 1:
    gradeOutput = "D+"

elif gradeInput == 1 or gradeInput >= 0.85:
    gradeOutput = "D"

elif gradeInput < 0.85 and gradeInput >= 0.7:
    gradeOutput = "D-"

else:
    gradeOutput = "F"

print("Your grade is:", gradeOutput)

File: Java-for-Everyone-Late-Objects-Horstmann/Chapter-3-Decisions/P3_13.java
/*Write a program that translates a number between 0 and 4 into the closest letter
grade. For example, the number 2.8 (which might have been the average of several
grades) would be converted to B–. Break ties in favor of the better grade; 
for example 2.85 should be a B.*/

import java.util.Scanner;

public class P3_13 {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		System.out.print("Numeric grade (0.0 - 4.0): ");
		double numeric_grade = input.nextDouble();
		input.close();
		String letter_grade = "F";
		if (numeric_grade > 4 || numeric_grade < 0) {
			System.out.println("Invalid grade. Grade must be between 0.0 and 4.0.");
			return;
		} else if (numeric_grade == 4 || numeric_grade >= 3.85) {
			letter_grade = "A";
		} else if (numeric_grade > 3.7) {
			letter_grade = "A-";
		} else if (numeric_grade > 3) {
			letter_grade = "B+";
		} else if (numeric_grade >= 2.85) {
			letter_grade = "B";
		} else if (numeric_grade >= 2.7) {
			letter_grade = "B-";
		} else if (numeric_grade > 2) {
			letter_grade = "C+";
		} else if (numeric_grade >= 1.85) {
			letter_grade = "C";
		} else if (numeric_grade >= 1.7) {
			letter_grade = "C-";
		} else if (numeric_grade > 1) {
			letter_grade = "D+";
		} else if (numeric_grade >= 0.85) {
			letter_grade = "D";
		} else if (numeric_grade >= 0.7) {
			letter_grade = "D-";
		} else {
			letter_grade = "F";
		}
		System.out.println("Your grade is: " + letter_grade);
	}
}
